src/core/libs/cardano_multiplatform_lib/src/tx_builder.rs

```diff
@@ -252,11 +253,46 @@ fn min_fee(tx_builder: &mut TransactionBuilder) -> Result<Coin, JsError> {
     // }
     let build = tx_builder.build()?;
     let full_tx = fake_full_tx(tx_builder, build)?;
-    fees::min_fee(
+    let base_fee = fees::min_fee(
         &full_tx,
         &tx_builder.config.fee_algo,
         &tx_builder.config.ex_unit_prices,
-    )
+    ).unwrap();
+
+    let mut reference_scripts_tier_price = BigNum::from(0);
+    if let Some(reference_inputs) = &tx_builder.reference_inputs.clone() {
```

Comment:
Include inputs as well as reference inputs? 

https://github.com/IntersectMBO/cardano-ledger/blob/7e65f0365eef647b9415e3fe9b3c35561761a3d5/eras/conway/impl/src/Cardano/Ledger/Conway/UTxO.hs#L155

Reply:
Timely: 
https://github.com/spacebudz/lucid/blob/a0e182fe77d553f6480d2b3ff5d1f38eee22e4e9/src/core/libs/cardano_multiplatform_lib/src/tx_builder.rs#L258

Reply:
Thanks!

Reply:
My patch
https://github.com/kompact-io/lucid/blob/2d0021dbd0605a53346885a81191e25f1ab3ba7b/src/core/libs/cardano_multiplatform_lib/src/tx_builder.rs#L262
(Could not find the right incantation to output the right wasm/js). 
This seems to be passing.

Reply:
Hey @waalge, I don't have access to your patch, but the upstream implementation you liked above seems to be wrong. It assumes a fixed cost per byte of 15 lovelace, when in fact the cost per byte increases with the total ref script size (the actual formula is `costPerByte = base * multiplier^(floor(costPerByte / range))`, where `base`, `multiplier` and `range` are new Conway-era protocol parameters.

Reply:
@groscoe Apologies, link should now be public. 

I tried running of lucids' branches and hit wasm issues. 

I think the code is slightly obscured by comment https://github.com/Liqwid-Labs/lucid/pull/1/files#r1711328007